runtime/cudaq/qis/managers/photonics/PhotonicsExecutionManager.cpp

```diff
@@ -151,242 +134,116 @@ class PhotonicsExecutionManager : public cudaq::BasicExecutionManager {
-            qpp::sample(shots, state, ids, sampleQudits.begin()->levels);
-        cudaq::ExecutionResult counts;
-        for (auto [result, count] : sampleResult) {
-          std::stringstream bitstring;
-          for (const auto &quditRes : result) {
-            bitstring << quditRes;
-          }
-          // Add to the sample result
-          // in mid-circ sampling mode this will append 1 bitstring
-          counts.appendResult(bitstring.str(), count);
-          // Reset the string.
-          bitstring.str("");
-          bitstring.clear();
-        }
-        executionContext->result.append(counts);
-      } else if (executionContext->name == "extract-state") {
-        executionContext->simulationState =
-            std::make_unique<cudaq::PhotonicsState>(
-                std::move(state), sampleQudits.begin()->levels);
-      }
-      // Reset the state and qudits
-      state.resize(0);
-      sampleQudits.clear();
+    if (!requestedAllocations.empty()) {
+      cudaq::info(
+          "[PhotonicsExecutionManager] Flushing remaining {} allocations "
+          "at handleExecutionContextEnded.",
+          requestedAllocations.size());
+      // If there are pending allocations, flush them to the simulator.
+      // Making sure the simulator's state is consistent with the number of
+      // allocations even though the circuit might be empty.
+      photonic_simulator()->allocateQudits(requestedAllocations.size());
+      requestedAllocations.clear();
     }
+    photonic_simulator()->resetExecutionContext();
   }
 
-  /// @brief Method for executing instructions.
   void executeInstruction(const Instruction &instruction) override {
-    auto operation = instructions[std::get<0>(instruction)];
-    operation(instruction);
+    flushRequestedAllocations();
+
+    // Get the data, create the Qudit* targets
+    auto [gateName, parameters, controls, targets, op] = instruction;
+
+    // Map the Qudits to Qubits
+    std::vector<std::size_t> localT;
+    std::transform(targets.begin(), targets.end(), std::back_inserter(localT),
+                   [](auto &&el) { return el.id; });
+    std::vector<std::size_t> localC;
+    std::transform(controls.begin(), controls.end(), std::back_inserter(localC),
+                   [](auto &&el) { return el.id; });
+
+    // Apply the gate
+    llvm::StringSwitch<std::function<void()>>(gateName)
+        .Case("plus", [&]() { photonic_simulator()->plus(localC, localT[0]); })
+        .Case("beam_splitter",
+              [&]() {
+                photonic_simulator()->beam_splitter(parameters[0], localC,
+                                                    localT);
+              })
+        .Case("phase_shift",
+              [&]() {
+                photonic_simulator()->phase_shift(parameters[0], localC,
+                                                  localT[0]);
+              })
+        .Default([&]() {
+          if (cudaq::customOpRegistry::getInstance().isOperationRegistered(
```

Comment:
Can you add tests to cover custom operations?

Reply:
Custom operation were removed in #[7da6e7d](https://github.com/NVIDIA/cuda-quantum/pull/2243/commits/7da6e7dbea71e2b1bb538b99864c7ae36e507842)